kml/serdes.cpp

```diff
@@ -455,43 +455,106 @@ void SaveTrackLayer(Writer & writer, TrackLayer const & layer,
   writer << indent << "<width>" << strings::to_string(layer.m_lineWidth) << "</width>\n";
 }
 
-void SaveTrackGeometry(Writer & writer, MultiGeometry const & geom)
+void SaveLineStrings(Writer & writer, MultiGeometry const & geom)
 {
-  size_t const sz = geom.m_lines.size();
-  if (sz == 0)
-  {
-    ASSERT(false, ());
-    return;
-  }
-
   auto linesIndent = kIndent4;
-  if (sz > 1)
+  auto const lineStringsSize = geom.GetNumberOfLinesWithouTimestamps();
+  
+  if (lineStringsSize > 1)
   {
     linesIndent = kIndent8;
     writer << kIndent4 << "<MultiGeometry>\n";
   }
 
-  for (auto const & e : geom.m_lines)
+  for (size_t lineIndex = 0; lineIndex < geom.m_lines.size(); ++lineIndex)
   {
-    if (e.empty())
+    auto const & line = geom.m_lines[lineIndex];
+    if (line.empty())
     {
-      ASSERT(false, ());
+      ASSERT(false, ("Unexpected empty Track"));
       continue;
     }
 
+    // Skip the tracks with the timestamps when writing the <LineString> points.
+    if (geom.HasTimestampsFor(lineIndex))
+      continue;
+
     writer << linesIndent << "<LineString><coordinates>";
 
-    writer << PointToString(e[0]);
-    for (size_t i = 1; i < e.size(); ++i)
-      writer << " " << PointToString(e[i]);
+    writer << PointToString(line[0]);
+    for (size_t pointIndex = 1; pointIndex < line.size(); ++pointIndex)
+      writer << " " << PointToString(line[pointIndex]);
 
     writer << "</coordinates></LineString>\n";
   }
 
-  if (sz > 1)
+  if (lineStringsSize > 1)
     writer << kIndent4 << "</MultiGeometry>\n";
 }
 
+void SaveGxTracks(Writer & writer, MultiGeometry const & geom)
+{
+  auto linesIndent = kIndent4;
+  auto const gxTracksSize = geom.GetNumberOfLinesWithTimestamps();
+  
+  if (gxTracksSize > 1)
+  {
+    linesIndent = kIndent8;
+    writer << kIndent4 << "<gx:MultiTrack>\n";
+    /// @TODO(KK): add the <altitudeMode>absolute</altitudeMode> if needed
+  }
+
+  for (size_t lineIndex = 0; lineIndex < geom.m_lines.size(); ++lineIndex)
+  {
+    auto const & line = geom.m_lines[lineIndex];
+    if (line.empty())
+    {
+      ASSERT(false, ("Unexpected empty Track"));
```

Comment:
This should fail in debug and print log in release:
```suggestion
      LOG(LERROR, ("Unexpected empty Track"));
```